lib/token-assert.js

```diff
@@ -77,18 +37,79 @@ TokenAssert.prototype.whitespaceBetween = function(options) {
  * @param {Boolean} [options.disallowNewLine=false]
  */
 TokenAssert.prototype.noWhitespaceBetween = function(options) {
+    options.exactly = 0;
+    this.spacesBetween(options);
+};
+
+/**
+ * Requires to have the whitespace between specified tokens with the provided options.
+ *
+ * @param {Object} options.token
+ * @param {Object} options.nextToken
+ * @param {String} [options.message]
+ * @param {Object} [options.atLeast] At least how many spaces the tokens are apart
+ * @param {Object} [options.atMost] At most how many spaces the tokens are apart
+ * @param {Object} [options.exactly] Exactly how many spaces the tokens are apart
+ * @param {Boolean} [options.disallowNewLine=false]
+ */
+TokenAssert.prototype.spacesBetween = function(options) {
     var token = options.token;
     var nextToken = options.nextToken;
-    if (nextToken.range[0] !== token.range[1] &&
-        (options.disallowNewLine || token.loc.end.line === nextToken.loc.start.line)
-    ) {
-        nextToken.whitespaceBefore = '';
+    var atLeast = options.atLeast;
+    var atMost = options.atMost;
+    var exactly = options.exactly;
+
+    if (!token || !nextToken) {
+        return;
+    }
+
+    this._validateOptions(options);
+
+    if (!options.disallowNewLine && token.loc.end.line !== nextToken.loc.start.line) {
+        return;
+    }
+
+    // Only attempt to remove or add lines if there are no comments between the two nodes
+    // as this prevents accidentally moving a valid token onto a line comment ed line
+    var fixed = this._file.getNextToken(options.token, {includeComments: true}) === nextToken;
+
+    var emitError = function(countPrefix, spaceCount) {
+        if (fixed) {
+            nextToken.whitespaceBefore = new Array(spaceCount + 1).join(' ');
+        }
+
+        var msgPostfix = token.value + ' and ' + nextToken.value;
+
+        if (!options.message) {
+            if (exactly === 0) {
+                // support noWhitespaceBetween
+                options.message = 'Unexpected whitespace between ' + msgPostfix;
+            } else if (exactly !== undefined) {
+                // support whitespaceBetween (spaces option)
+                options.message = spaceCount + ' spaces required between ' + msgPostfix;
+            } else if (atLeast === 1 && atMost === undefined) {
```

Comment:
Maybe switch-case here?
Is it just a BC or what about `atLeast === undefined && atMost === 1` ?
Probably better to use `typeof variable === 'undefined'`.